langchain/src/agents/openai/index.ts

```diff
@@ -199,28 +207,49 @@ export class OpenAIAgent extends Agent {
     }
 
     // Split inputs between prompt and llm
-    const llm = this.llmChain.llm as ChatOpenAI;
+    const llm = this.llmChain.llm as
+      | ChatOpenAI
+      | Runnable<
+          BaseLanguageModelInput,
+          BaseMessageChunk,
+          ChatOpenAICallOptions
+        >;
+
     const valuesForPrompt = { ...newInputs };
-    const valuesForLLM: (typeof llm)["CallOptions"] = {
+    const valuesForLLM: CallOptionsIfAvailable<typeof llm> = {
       functions: this.tools.map(formatToOpenAIFunction),
     };
     const callKeys =
       "callKeys" in this.llmChain.llm ? this.llmChain.llm.callKeys : [];
     for (const key of callKeys) {
       if (key in inputs) {
-        valuesForLLM[key as keyof (typeof llm)["CallOptions"]] = inputs[key];
+        valuesForLLM[key as keyof CallOptionsIfAvailable<typeof llm>] =
+          inputs[key];
         delete valuesForPrompt[key];
       }
     }
 
-    const promptValue = await this.llmChain.prompt.formatPromptValue(
-      valuesForPrompt
-    );
-    const message = await llm.predictMessages(
-      promptValue.toChatMessages(),
-      valuesForLLM,
-      callbackManager
-    );
+    const promptValue =
+      await this.llmChain.prompt.formatPromptValue(valuesForPrompt);
+
+    let message: BaseMessage;
+    if (Runnable.isRunnable(llm)) {
```

Comment:
All llms extend runnables so this if statement won't work in the way it's intended to.

Instead you should be checking for a class property which would identify the openai LLM, and else would be runnable

Reply:
Good point, I think we don't need the branch at all then if it's still working. Will remove.